### PR TITLE
fix: remove FAB rendered menu in favor of react based one

### PR DESCRIPTION
**Pull request by nytai**
### SUMMARY

mostly what the title says. 

### TEST PLAN

Nothing should change from a user perspective.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by mistercrunch** [comment]
years

**Comment by rusackas** [comment]
excited

**Comment by rusackas** [comment]
Tempted to just approve/merge this puppy, but hoping to see if @dpgaspar has any input on it first.

File: superset/views/base.py [hunk 1]
```diff
@@ -256,10 +256,13 @@ def json_response(
 
 
 def menu_data() -> Dict[str, Any]:
-    menu = appbuilder.menu.get_data()
+    menu = []
     root_path = "#"
     logo_target_path = ""
-    if not g.user.is_anonymous:
+    user_is_anonymous = True
+    if hasattr(g, "user") and not g.user.is_anonymous:
```
Comment (dpgaspar) [inline superset/views/base.py:263]: This defeats the `Public` role. `menu.get_data` is already checking if the user is authenticated and if not getting permissions from the `Public` role. Is there any other blocker?

**Review by dpgaspar — COMMENTED** [review]
Looks good, but we should make sure that the public role still works

File: superset/views/base.py (hunk 1 above)
Comment (nytai) [reply superset/views/base.py:263]: I think the main issue is that the public role doesn't work. I was getting an error on the sign in view for `menu.get_data`.

Also this test https://github.com/preset-io/incubator-superset/blob/master/tests/schedules_test.py#L495 was failing without the `hasattr(g, "user")` check.

**Comment by rusackas** [comment]
When this PR is merged, https://github.com/apache/incubator-superset/pull/10423 is ready to follow.

File: superset/views/base.py (hunk 1 above)
Comment (dpgaspar) [reply superset/views/base.py:263]: Isn't that a flaky test? would risk to say it was a coincidence. 
Re checked and it works with anonymous and the public role, https://github.com/dpgaspar/Flask-AppBuilder/blob/master/flask_appbuilder/tests/test_menu.py#L101 but we need to add the endpoint to the public role. Yet `menu.get_data` shoud work

**Review by rusackas — APPROVED** [review]

**Comment by rusackas** [comment]
Impacts #8976

**Pull request merged**